subprojects/core/src/test/groovy/org/gradle/internal/jpms/JavaModuleDetectorTest.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.jpms
+
+import org.gradle.api.file.FileCollection
+import org.gradle.api.internal.file.AbstractFileCollection
+import org.gradle.api.jpms.ModularClasspathHandling
+import org.gradle.cache.internal.TestFileContentCacheFactory
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.util.JarUtils
+import org.junit.Rule
+import spock.lang.Specification
+
+class JavaModuleDetectorTest extends Specification {
+
+    @Rule
+    TestNameTestDirectoryProvider tmpDir = new TestNameTestDirectoryProvider()
+
+    TestFileContentCacheFactory cacheFactory = new TestFileContentCacheFactory()
+    JavaModuleDetector moduleDetector = new JavaModuleDetector(cacheFactory)
+    ModularClasspathHandling modularClasspathHandling = new DefaultModularClasspathHandling()
+
+    def setup() {
+        modularClasspathHandling.inferModulePath = true
+    }
+
+    def "detects modules on classpath"() {
+        def path = path('lib.jar', 'module.jar', 'classes', 'classes-module', 'automaticModule.jar')
```

Comment:
I was thinking about doing this as follow up.

But you are right. Even if we do not yet support building MR Jar in any specific way, we should consume them correctly here. As `javac` and `java`  would already "consume" them the right way automatically.

I will add support for it to the detector.

Reply:
I added very basic support now. It looks if there is a "module-info" for _any_ java version.

I think the best solution would be to record for which java versions it is a module so that we can decide based on the version we are compiling for.

But then we need to cache more information. Not just true/false. Which we could.

I wonder though, how practical that is. Why should I have something that is **not** a module in Java 9, but one in Java 10...